### PR TITLE
Catalog controller

### PR DESCRIPTION
**Pull request by arbulu89**
Add catalog controller to phoenix. The checks are loaded from the file system by now. Storing and loading the checks in the database should be almost a plug and play thing, so it can be added later on pretty easily

### CODE REVIEW

File: lib/wanda_web/views/catalog_view.ex [hunk 1]
```diff
@@ -0,0 +1,11 @@
+defmodule WandaWeb.CatalogView do
+  use WandaWeb, :view
+
+  alias Wanda.Catalog.Check
+
+  def render("list_catalog.json", %{catalog: catalog}) do
+    %{data: render_many(catalog, WandaWeb.CatalogView, "check.json", as: :check)}
```
Comment (dottorblaster) [inline lib/wanda_web/views/catalog_view.ex:7]: Same as above :smile:

**Review by dottorblaster — CHANGES_REQUESTED** [review]
Just a tiny thing then I think we're good

File: lib/wanda_web/schemas/catalog_response.ex [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  alias OpenApiSpex.Schema
+  alias WandaWeb.Schemas.CatalogCheck
+
+  OpenApiSpex.schema(%{
+    title: "CatalogResponse",
+    description: "Checks catalog listing response",
+    type: :object,
+    properties: %{
+      data: %Schema{type: :array, description: "List of catalog checks", items: CatalogCheck}
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply lib/wanda_web/schemas/catalog_response.ex]: Done!

**Review by dottorblaster — APPROVED** [review]
LGTM

**Comment by dottorblaster** [comment]
image

**Review by nelsonkopliku — CHANGES_REQUESTED** [review]
Looks good, just a couple of comments.

**Review by rtorrero — APPROVED** [review]
Thanks @arbulu89! It seems to be ready to merge for me, but I also subscribe what Nelson says in his last comment

File: test/catalog_test.exs [hunk 3]
```diff
@@ -12,10 +12,25 @@ defmodule Wanda.CatalogTest do
...
   describe "checks catalog" do
+    test "should return the whole catalog" do
+      catalog_path = Application.fetch_env!(:wanda, Wanda.Catalog)[:catalog_path]
+      {:ok, files} = File.ls(catalog_path)
+
+      catalog = Catalog.get_catalog()
+      assert length(files) == length(catalog)
+
+      Enum.each(catalog, fn check ->
+        assert %Check{} = check
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply test/catalog_test.exs:23]: Well, the signature is simply a spec, it doesn't block you to send other types (nor credo or dialyzer complain as far as I know). Having incorrect signatures is more common than what we expect.
Anyway, I will try to think how I can improve it

[the remark this answers is not in the document]
Comment (arbulu89) [reply test/catalog_test.exs]: @nelsonkopliku Test change

[the remark this answers is not in the document]
Comment (nelsonkopliku) [reply test/catalog_test.exs]: Alright looks good, thanks!

**Review by nelsonkopliku — APPROVED** [review]
A lil conflict to fix, for the rest green light.

**Pull request merged**